src/EnergyPlus/ExhaustAirSystemManager.cc

```diff
@@ -0,0 +1,966 @@
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// ObjexxFCL Headers
+#include <ObjexxFCL/Array.functions.hh>
+#include <ObjexxFCL/Fmath.hh>
+
+// EnergyPlus Headers
+#include <AirflowNetwork/Elements.hpp>
+#include <EnergyPlus/BranchNodeConnections.hh>
+#include <EnergyPlus/Data/EnergyPlusData.hh>
+#include <EnergyPlus/DataContaminantBalance.hh>
+#include <EnergyPlus/DataEnvironment.hh>
+#include <EnergyPlus/DataHVACGlobals.hh>
+#include <EnergyPlus/DataHeatBalFanSys.hh>
+#include <EnergyPlus/DataHeatBalance.hh>
+#include <EnergyPlus/DataIPShortCuts.hh>
+#include <EnergyPlus/DataLoopNode.hh>
+#include <EnergyPlus/DataSizing.hh>
+#include <EnergyPlus/DataZoneEquipment.hh>
+#include <EnergyPlus/ExhaustAirSystemManager.hh>
+#include <EnergyPlus/Fans.hh>
+#include <EnergyPlus/GeneralRoutines.hh>
+#include <EnergyPlus/HVACFan.hh>
+#include <EnergyPlus/InputProcessing/InputProcessor.hh>
+#include <EnergyPlus/MixerComponent.hh>
+#include <EnergyPlus/NodeInputManager.hh>
+// #include <EnergyPlus/ReturnAirPathManager.hh> //2022-01-14: replace with exhaust system
+#include <EnergyPlus/ScheduleManager.hh>
+#include <EnergyPlus/UtilityRoutines.hh>
+// #include <EnergyPlus/ZonePlenum.hh> //2022-01-14: may not needed for this exhaust system
+
+namespace EnergyPlus {
+
+namespace ExhaustAirSystemManager {
+    // Module containing the routines dealing with the AirLoopHVAC:ExhaustSystem
+
+    // struct MixerBranchZone
+    //{
+    //    int mixerInletNodeNum;
+    //    int zoneExhaustNodeNum;
+    //    int zoneNum;
+    //    bool collected;
+
+    //    // default constructor
+    //    // Question: why the constructor skipped the first element std::string Name?
+    //    MixerBranchZone() : mixerInletNodeNum(0), zoneExhaustNodeNum(0), zoneNum(0), collected(false)
+    //    {
+    //    }
+
+    //    MixerBranchZone(int a, int b, int c, bool d) : mixerInletNodeNum(a), zoneExhaustNodeNum(b), zoneNum(c), collected(d)
+    //    {
+    //    }
+    //};
+
+    // std::vector<MixerBranchZone> mixerToZoneTable;
+    // map might be aa better choice:
+    std::map<int, int> mixerBranchMap;
+    std::map<int, int> mixerIndexMap;
+
+    bool mappingDone = false;
+
+    void SimExhaustAirSystem(EnergyPlusData &state, bool FirstHVACIteration)
+    {
+        // Locals
+        int ExhaustAirSystemNum;
+
+        // Obtains and Allocates Mixer related parameters from input file
+        // 2022-01-14: To do: Need to define a correponding state.dataExhAirSystemMgr data structure
+        if (state.dataExhAirSystemMrg->GetInputFlag) { // First time subroutine has been entered
+            GetExhaustAirSystemInput(state);
+            state.dataExhAirSystemMrg->GetInputFlag = false;
+        }
+
+        for (ExhaustAirSystemNum = 1; ExhaustAirSystemNum <= state.dataZoneEquip->NumExhaustAirSystems; ++ExhaustAirSystemNum) {
```

Comment:
Declare the loop induction variable in loop scope, e.g., `for (int ExhaustAirSystemNum = 1; ...)`.

Reply:
Thanks. The change is made for this local variable to reduce the scope.